### PR TITLE
Fix compile warnings in "ide"

### PR DESCRIPTION
**Pull request by bmuskalla**
A lot of the `@SuppressWarnings` are required because
FileContentMerger/XmlFileContentMerger use MutableActionSet
without a type (even though it is typed). Given we expose the action
sets via getBeforeMerged/getWhenMerged, we can't make it typed without
breaking API compatibility.

### CODE REVIEW

**Comment by bmuskalla** [comment]
@bot-gradle test RFM

**Comment by bmuskalla** [comment]
@donat I thought you might be in the better position to review this (was part of @jjohannes move to the strictly plugin). Let me know if you don't have the capacity to review this.

File: subprojects/ide/src/main/java/org/gradle/plugins/ide/idea/model/Project.java [hunk 1]
```diff
@@ -321,6 +322,11 @@ private Node findOrCreateLibraryTable() {
         return libraryTable;
     }
 
+    private static void setNodeAttribute(Node node, String key, Object value) {
```
Comment (donat) [inline subprojects/ide/src/main/java/org/gradle/plugins/ide/idea/model/Project.java:325]: Extract this to a separate class and remove the duplication.

File: subprojects/ide/src/main/java/org/gradle/plugins/ide/idea/internal/IdeaScalaConfigurer.java [hunk 2]
```diff
@@ -232,7 +237,7 @@ private VersionNumber findIdeaTargetVersion() {
         if (targetVersionString != null) {
             targetVersion = VersionNumber.parse(targetVersionString);
             if (targetVersion.equals(VersionNumber.UNKNOWN)) {
-                throw new GradleScriptException("String \'" + targetVersionString + "\' is not a valid value for IdeaModel.targetVersion.", null);
```
Comment (donat) [inline subprojects/ide/src/main/java/org/gradle/plugins/ide/idea/internal/IdeaScalaConfigurer.java:235]: Nitpick: this is an unrelated change.

File: subprojects/ide/src/main/java/org/gradle/plugins/ide/idea/model/Project.java (hunk 1 above)
Comment (bmuskalla) [reply subprojects/ide/src/main/java/org/gradle/plugins/ide/idea/model/Project.java:325]: This was actually on purpose to avoid a random one-line utility method that most people dealing with such code will not remember. Given its an oneliner in three places, I inlined it. But no strong opinion. How/where would you put something like that?

File: subprojects/ide/src/main/java/org/gradle/plugins/ide/idea/internal/IdeaScalaConfigurer.java (hunk 2 above)
Comment (bmuskalla) [reply subprojects/ide/src/main/java/org/gradle/plugins/ide/idea/internal/IdeaScalaConfigurer.java:235]: It's a warning in our current intellij settings (the escaping is unnecessary)

File: subprojects/ide/src/main/java/org/gradle/plugins/ide/idea/model/Project.java (hunk 1 above)
Comment (donat) [reply subprojects/ide/src/main/java/org/gradle/plugins/ide/idea/model/Project.java:325]: It's fine like this, I just wanted to point out that it might make sense to change it.

File: subprojects/ide/src/main/java/org/gradle/plugins/ide/idea/internal/IdeaScalaConfigurer.java (hunk 2 above)
Comment (donat) [reply subprojects/ide/src/main/java/org/gradle/plugins/ide/idea/internal/IdeaScalaConfigurer.java:235]: No, it's me. I though you actually removed the single quotes. Ignore my comment, all good :)
